cmd/flipt/main.go

```diff
@@ -147,69 +100,97 @@ func main() {
 		Date:      date,
 		GoVersion: goVersion,
 	}); err != nil {
-		logger().Fatal("executing template", zap.Error(err))
+		fatal("executing template", zap.Error(err))
 	}
 
 	banner = buf.String()
 
-	cobra.OnInitialize(func() {
```

Comment:
I popped cobra to have a loop. `initializers` run just before `Run` or `RunE`, but after flag parsing.

There are some bits of validating that go on between invoking the run function and after calling this function.
However, I can't necessarily see that causing problems, since we don't seem to use half the things which get triggered (e.g. `PreRun` and so on), let alone have a dependency order which is dictated by these:
https://github.com/spf13/cobra/blob/v1.6.1/command.go#L876-L921

Running the same process at the start of `Run` / `RunE` should be equivalent in our situation.
Allowing us to not dictate that every function has to parse a `config.yaml` in a particular location in order to be invoked. Which is what this did.

I was going to just make the default of `--config` be empty string and switch on that. But then realized that would actually be a breaking change if folks are relying on just editing the default yaml file at the usual location and ommitting the flag.